builtin/providers/google/resource_sql_user.go

```diff
@@ -77,6 +83,8 @@ func resourceSqlUserCreate(d *schema.ResourceData, meta interface{}) error {
 			"user %s into instance %s: %s", name, instance, err)
 	}
 
+	d.SetId(fmt.Sprintf("%s.%s", instance, name))
```

Comment:
So I take back part of what I told you in person. We shouldn't change the format of the ID, since that will break refreshing (since the ID in state will still be the old version). At this point we have three possible solutions:
1. Write a migration function to change all the IDs in state. I _think_ it would work, but migration functions are kind of a pain.
2. Do the ID parsing in the StateFn, so instead of doing a `schema.ImportStatePassthrough`, create a custom function that parses the ID and then sets the two properties in the state before it gets passed to the read fn.
3. Create a custom function that reads all available instances, and then tries each one to see if the user belongs to that instance (see [resource_pagerduty_service_integration](https://github.com/hashicorp/terraform/blob/master/builtin/providers/pagerduty/resource_pagerduty_service_integration.go) for an example). This is actually less bad than I initially made it out to be. Since Terraform is currently storing the user name as the ID, it's already not allowing multiple users with the same name in different instances. And so a naive import should fail in this case anyway. However the point could be made that Terraform _should_ allow users with the same name in different instances.

I'm not sure yet which the best solution is. I'll consult with Hashicorp people and get back to you.